### PR TITLE
Added search in properties of a card

### PR DESCRIPTION
**Pull request by BharatKalluri**
#### Summary

Search now also searches in the properties of the card along with the name. 

#### Ticket Link

https://github.com/mattermost/focalboard/issues/128

### CODE REVIEW

File: webapp/src/viewModel/boardTree.ts [hunk 1]
```diff
@@ -198,7 +198,13 @@ class MutableBoardTree implements BoardTree {
             return cards.slice()
         }
 
-        return cards.filter((card) => card.title?.toLocaleLowerCase().indexOf(searchText) !== -1)
+        return cards.filter((card: Card) => {
+            const searchTextInCardTitle: boolean = card.title?.toLocaleLowerCase().includes(searchText)
+            const searchTextInProperties: boolean = Object.values(card.properties).
+                map((el: string) => el.toLowerCase().includes(searchText)).
+                some((el: boolean) => el)
```
Comment (chenilim) [inline webapp/src/viewModel/boardTree.ts:205]: * Replace this with a forEach loop and return true early on a match
* You should also be able to write `card.properties.forEach(...`

**Review by chenilim — CHANGES_REQUESTED** [review]
Thanks for the PR @BharatKalluri! A couple perf improvement requests.

File: webapp/src/viewModel/boardTree.ts [hunk 2]
```diff
@@ -198,7 +198,13 @@ class MutableBoardTree implements BoardTree {
             return cards.slice()
         }
 
-        return cards.filter((card) => card.title?.toLocaleLowerCase().indexOf(searchText) !== -1)
+        return cards.filter((card: Card) => {
+            const searchTextInCardTitle: boolean = card.title?.toLocaleLowerCase().includes(searchText)
```
[the remark this answers is not in the document]
Comment (BharatKalluri) [reply webapp/src/viewModel/boardTree.ts:202]: Agreed 👍🏽 
Made changes @ https://github.com/mattermost/focalboard/pull/222/commits/0b02ddfc4bf7bb4a42ae510e20d196405d846232

**Review by chenilim — APPROVED** [review]
Thanks for the updated @BharatKalluri! I realized there are a few more subtleties with handling select property types. I'll merge this PR and edit. Thank you for the contribution!

**Pull request merged**